test/e2e/init_scale_zero_test.go

```diff
@@ -0,0 +1,313 @@
+	}
+
+	// Verify that no pods are created
+	pods := clients.KubeClient.Kube.CoreV1().Pods(test.ServingNamespace)
+	podList, err := pods.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", serving.RevisionLabelKey, objects.Revision.Name),
+	})
+	if err != nil {
+		t.Fatalf("Failed to list all pods: %v: %v", names.Service, err)
+	}
+	if len(podList.Items) != 0 {
+		t.Fatalf("Did not scale to 0; %d pods are created.", len(podList.Items))
+	}
+
+	t.Log("Creating a new Service with check validity on deploy being true.")
+	names = test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "helloworld",
+	}
+	objects, _, err = v1a1test.CreateRunLatestServiceReady(t, clients, &names,
+		false, /* https TODO(taragu) turn this on after helloworld test running with https */
+		v1a1testing.WithConfigAnnotations(map[string]string{
+			autoscaling.CheckValidityOnDeployAnnotation: "true",
+		}))
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+
+	// Verify that pods are created
+	pods = clients.KubeClient.Kube.CoreV1().Pods(test.ServingNamespace)
+	podList, err = pods.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", serving.RevisionLabelKey, objects.Revision.Name),
+	})
+	if err != nil {
+		t.Fatalf("Failed to list all pods: %v: %v", names.Service, err)
+	}
+	if len(podList.Items) == 0 {
+		t.Fatal("No pods are created.")
+	}
+}
+
+// TestInitScaleZeroClusterLevel tests setting of annotation checkValidityOnDeploy on
+// the cluster level
+func TestInitScaleZeroClusterLevel(t *testing.T) {
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := Setup(t)
+	namesCheckValidityUnset := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "helloworld",
+	}
+	namesCheckValidityFalse := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "helloworld",
+	}
+	namesCheckValidityTrue := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "helloworld",
+	}
+
+	test.CleanupOnInterrupt(func() {
+		test.TearDown(clients, namesCheckValidityUnset)
+		test.TearDown(clients, namesCheckValidityFalse)
+		test.TearDown(clients, namesCheckValidityTrue)
+
+	})
+	defer test.TearDown(clients, namesCheckValidityUnset)
+	defer test.TearDown(clients, namesCheckValidityFalse)
+	defer test.TearDown(clients, namesCheckValidityTrue)
+
+	t.Log("Fetch config-defaults ConfigMap")
+	configDefaults, err := defaultCM(clients)
+	if err != nil || configDefaults == nil {
+		t.Fatalf("Error fetching config-defaults: %v", err)
+	} else {
+		var newCM *v1.ConfigMap
+		configDefaults.Data["check-validity-on-deploy"] = "false"
+		if newCM, err = patchDefaultCM(clients, configDefaults); err != nil {
+			t.Fatalf("Failed to update config defaults: %v", err)
+		}
+		t.Logf("Successfully updated configMap.")
+		newCM.Data["check-validity-on-deploy"] = "true"
+		test.CleanupOnInterrupt(func() { restoreClusterDefaults(t, clients, newCM) })
```

Comment:
patch it back to the original using the first `configDefaults`

Reply:
I'm getting this error if I try to use the first configDefaults: `Operation cannot be fulfilled on configmaps "config-defaults": the object has been modified; please apply your changes to the latest version and try again`